msbuild/Xamarin.MacDev.Tasks/Tasks/ResolveNativeReferences.cs

```diff
@@ -284,6 +273,21 @@ static bool TryGetInfoPlist (TaskLoggingHelper log, string resourcePath, string
 			return true;
 		}
 
+		void SetMetadataNativeLibrary (ITaskItem item, string nativeLibraryPath)
+		{
+			item.ItemSpec = GetActualLibrary (nativeLibraryPath);
+			if (item.ItemSpec.EndsWith (".a", StringComparison.OrdinalIgnoreCase)) {
+				item.SetMetadata ("Kind", "Static");
+				item.SetMetadata ("PublishFolderType", "StaticLibrary");
+			} else if (item.ItemSpec.EndsWith (".dylib", StringComparison.OrdinalIgnoreCase)) {
+				item.SetMetadata ("Kind", "Dynamic");
+				item.SetMetadata ("PublishFolderType", "DynamicLibrary");
+			} else {
+				item.SetMetadata ("Kind", "Framework");
+				item.SetMetadata ("PublishFolderType", "AppleFramework");
+			}
```

Comment:
we could simplify that a little:
```suggestion
			// defaults
			var kind = "Framework";
			var folderType = "AppleFramework";
			
			if (item.ItemSpec.EndsWith (".a", StringComparison.OrdinalIgnoreCase)) {
				kind = "Static";
				folderType = "StaticLibrary);
			}
			
			if (item.ItemSpec.EndsWith (".dylib", StringComparison.OrdinalIgnoreCase)) {
				kind = "Dynamic";
				folderType = "DynamicLibrary";
			}
			
			item.SetMetadata ("Kind", kind);
			item.SetMetadata ("PublishFolderType", folderType); 

```
Not a blocker to land, just a proposal.